src/libutil/util.cc

```diff
@@ -389,12 +394,20 @@ void writeLine(int fd, string s)
 }
 
 
-static void _deletePath(const Path & path, unsigned long long & bytesFreed)
+static void _deletePath(int parentfd, const Path & path, unsigned long long & bytesFreed)
 {
     checkInterrupt();
 
+    string name(baseNameOf(path));
+
+    AutoCloseFD fd = openat(parentfd, name.c_str(), 0);
```

Comment:
May be a good idea to use `O_NOFOLLOW` here.

Reply:
Hit this while I was trying to fix the installcheck :D